module/Frontpage/src/Controller/PollAdminController.php

```diff
@@ -57,30 +80,52 @@ public function listAction(): ViewModel
     /**
      * Approve a poll.
      */
-    public function approveAction(): Response
+    public function approveAction(): ViewModel|Response
     {
+        if (!$this->aclService->isAllowed('approve', 'poll')) {
+            throw new NotAllowedException($this->translator->translate('You are not allowed to approve polls'));
+        }
+
         if ($this->getRequest()->isPost()) {
             $pollId = $this->params()->fromRoute('poll_id');
             $poll = $this->pollService->getPoll($pollId);
-            $this->pollService->approvePoll($poll, $this->getRequest()->getPost());
 
-            return $this->redirect()->toRoute('admin_poll');
+            if (null !== $poll) {
+                $approvalForm = $this->pollService->getPollApprovalForm();
+                $approvalForm->bind($poll);
+                $approvalForm->setData($this->getRequest()->getPost()->toArray());
+
+                if ($approvalForm->isValid()) {
```

Comment:
Why is this validation necessary? And have you tested the changes?

Reply:
It is to validate the expiry date, there is not really a reason to actually `bind` the model but I just moved the code from the service (and it still works).